crates/bevy_platform/src/cfg.rs

```diff
@@ -0,0 +1,226 @@
+///
+/// The second advantage is a particularly powerful tool, as it allows consuming crates to use
+/// functionality in a defining crate regardless of what crate in the dependency graph enabled the
+/// relevant feature.
+///
+/// For example, consider a crate `foo` that depends on another crate `bar`.
+/// `bar` has a feature "`faster_algorithms`".
+/// If `bar` defines a "`faster_algorithms`" alias:
+///
+/// ```ignore
+/// define_alias! {
+///     feature = "faster_algorithms" => { faster_algorithms }
+/// }
+/// ```
+///
+/// Now, `foo` can gate its usage of those faster algorithms on the alias, avoiding the need to
+/// expose its own "`faster_algorithms`" feature.
+/// This also avoids the unfortunate situation where one crate activates "`faster_algorithms`" on
+/// `bar` without activating that same feature on `foo`.
+///
+/// Once an alias is defined, there are 4 ways you can use it:
+///
+/// 1. Evaluate with no contents to return a `bool` indicating if the alias is active.
+///    ```rust
+///    # use bevy_platform::cfg;
+///    if cfg::std!() {
+///        // Have `std`!
+///    } else {
+///        // No `std`...
+///    }
+///    ```
+/// 2. Pass a single code block which will only be compiled if the alias is active.
+///    ```rust
+///    # use bevy_platform::cfg;
+///    cfg::std! {
+///        // Have `std`!
+///    # ()
+///    }
+///    ```
+/// 3. Pass a single `if { ... } else { ... }` expression to conditionally compile either the first
+///    or the second code block.
+///    ```rust
+///    # use bevy_platform::cfg;
+///    cfg::std! {
+///        if {
+///            // Have `std`!
+///        } else {
+///            // No `std`...
+///        }
+///    }
+///    ```
+/// 4. Use in a [`switch`] arm for more complex conditional compilation.
+///    ```rust
+///    # use bevy_platform::cfg;
+///    cfg::switch! {
+///        cfg::std => {
+///            // Have `std`!
+///        }
+///        cfg::alloc => {
+///            // No `std`, but do have `alloc`!
+///        }
+///        _ => {
+///            // No `std` or `alloc`...
+///        }
+///    }
+///    ```
+#[doc(inline)]
+pub use crate::define_alias;
+
+#[doc(hidden)]
+#[macro_export]
+macro_rules! switch {
+    ({ $($tt:tt)* }) => {{
+        $crate::switch! { $($tt)* }
+    }};
+    (_ => { $($output:tt)* }) => {
+        $($output)*
+    };
+    (
+        $cond:path => $output:tt
+        $($( $rest:tt )+)?
+    ) => {
+        $cond! {
+            if {
+                $crate::switch! { _ => $output }
+            } else {
+                $(
+                    $crate::switch! { $($rest)+ }
+                )?
+            }
+        }
+    };
+    (
+        #[cfg($cfg:meta)] => $output:tt
+        $($( $rest:tt )+)?
+    ) => {
+        #[cfg($cfg)]
+        $crate::switch! { _ => $output }
+        $(
+            #[cfg(not($cfg))]
+            $crate::switch! { $($rest)+ }
+        )?
+    };
+}
+
```

Comment:
```suggestion

/// A macro for when a feature is disabled.
///
/// When called like a function with no arguments, this will always return `false`.
/// When called with a block of code, it will never run that code.
/// When called if an `if { ... } else { ... }`, it will always run the second (`else`) block.
```

You may want to consider renaming this macro as well, as it's not a no-op when `if { ... } else { ... }` is used. Perhaps `feature_disabled`?

Reply:
I've gone for `enabled` and `disabled`, since you can actually use this for any `cfg(...)` statement, not just features.